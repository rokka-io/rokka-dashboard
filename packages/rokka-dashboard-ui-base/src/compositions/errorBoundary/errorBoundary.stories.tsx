import { boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React, { FunctionComponent } from 'react';
import { ErrorBoundary } from './errorBoundary';

interface ErrorThrowingComponentProps {
  throwError?: boolean;
}

const ErrorThrowingComponent: FunctionComponent<ErrorThrowingComponentProps> = ({ throwError = false }) => {
  if (throwError) {
    throw new Error('I error always');
  }
  return <span>some text here</span>;
};

storiesOf('Compositions', module).add('ErrorBoundary', () => (
  <ErrorBoundary>
    <ErrorThrowingComponent throwError={boolean('throw error', false)} />
  </ErrorBoundary>
));
