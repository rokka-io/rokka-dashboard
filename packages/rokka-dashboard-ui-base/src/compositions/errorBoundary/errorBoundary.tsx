import React, { Component, ErrorInfo, ReactNode } from 'react';
import { ErrorPage } from '../../components/errorPage/errorPage';

interface ErrorBoundaryProps {
  children: ReactNode;
}
interface ErrorBoundaryState {
  hasError: boolean;
}

export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
  public state = { hasError: false };

  public componentDidCatch(error: Error, info: ErrorInfo) {
    if (process.env.NODE_ENV !== 'test') {
      // tslint:disable-next-line:no-console
      console.error(error, info);
    }

    this.setState({
      hasError: true
    });
  }

  public render() {
    if (this.state.hasError) {
      return <ErrorPage />;
    }
    return this.props.children;
  }
}
