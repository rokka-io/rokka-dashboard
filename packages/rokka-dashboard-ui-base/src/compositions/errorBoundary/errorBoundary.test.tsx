import { mount } from 'enzyme';
import React from 'react';

import { ErrorBoundary } from './errorBoundary';

const SomeComponent = () => {
  return <span>SomeComponent's text</span>;
};

describe('Compositions', () => {
  describe('ErrorBoundary', () => {
    it('renders the child if no error happened', () => {
      const component = mount(<ErrorBoundary>test</ErrorBoundary>);

      expect(component).toMatchSnapshot();
    });
    it('renders the error page if an error happened', () => {
      const component = mount(
        <ErrorBoundary>
          <SomeComponent />
        </ErrorBoundary>
      );
      component.find(SomeComponent).simulateError(new Error('an error happened'));

      expect(component).toMatchSnapshot();
    });
  });
});
