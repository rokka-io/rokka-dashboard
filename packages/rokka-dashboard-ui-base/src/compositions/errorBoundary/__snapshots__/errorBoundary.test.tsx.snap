// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compositions ErrorBoundary renders the child if no error happened 1`] = `
<ErrorBoundary>
  test
</ErrorBoundary>
`;

exports[`Compositions ErrorBoundary renders the error page if an error happened 1`] = `
<ErrorBoundary>
  <Component>
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-kkGfuU",
              "isStatic": true,
              "lastClassName": "kfmBKs",
              "rules": Array [
                "
  background-color: ",
                "#fff",
                ";
  padding: ",
                "1rem",
                ";

  :after {
    content: '';
    display: table;
    clear: both;
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-kkGfuU",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-kkGfuU kfmBKs"
        >
          <styled.h1>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bZQynM",
                    "isStatic": false,
                    "lastClassName": "fCytKC",
                    "rules": Array [
                      "
  font-family: ",
                      "\\"Liip Etica Book\\", sans-serif",
                      ";
  font-size: ",
                      "1.5rem",
                      ";
  color: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.h1",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bZQynM",
                  "target": "h1",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <h1
                className="sc-bZQynM fCytKC"
              >
                Something went wrong
              </h1>
            </StyledComponent>
          </styled.h1>
          <p>
            Unable to load the page
            <span
              aria-label="sad"
              role="img"
            >
              üòü
            </span>
            . Please
             
            <styled.a
              href="https://github.com/rokka-io/rokka-dashboard"
              rel="noopener"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-gqjmRU",
                      "isStatic": true,
                      "lastClassName": "dEOasl",
                      "rules": Array [
                        "
  font-family: ",
                        "\\"Liip Etica Book\\", sans-serif",
                        ";
  color: ",
                        "#6EA644",
                        ";
  word-wrap: break-word;
  text-decoration: underline;

  :hover {
    text-decoration: none;
  }
",
                      ],
                    },
                    "displayName": "styled.a",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-gqjmRU",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                href="https://github.com/rokka-io/rokka-dashboard"
                rel="noopener"
              >
                <a
                  className="sc-gqjmRU dEOasl"
                  href="https://github.com/rokka-io/rokka-dashboard"
                  rel="noopener"
                >
                  file an issue on GitHub.
                </a>
              </StyledComponent>
            </styled.a>
            Thanks!
             
            <span
              aria-label="yay"
              role="img"
            >
              üòç
            </span>
          </p>
        </div>
      </StyledComponent>
    </styled.div>
  </Component>
</ErrorBoundary>
`;
